packages/contracts-bedrock/test/properties/medusa/ProtocolAtomic.t.sol

```diff
@@ -0,0 +1,199 @@
+        } catch {
+            assert(address(destinationToken) == address(sourceToken) || sourceBalanceBefore < amount);
+        }
+    }
+
+    /// @notice pick one already-deployed supertoken and mint an arbitrary amount of it
+    /// necessary so there is something to be bridged :D
+    /// TODO: will be replaced when testing the factories and `convert()`
+    function fuzz_MintSupertoken(uint256 index, uint96 amount) external {
+        index = bound(index, 0, allSuperTokens.length - 1);
+        address addr = allSuperTokens[index];
+        vm.prank(BRIDGE);
+        // medusa calls with different senders by default
+        OptimismSuperchainERC20(addr).mint(msg.sender, amount);
+        // currentValue will be zero if key is not present
+        (,uint256 currentValue) = ghost_totalSupplyAcrossChains.tryGet(MESSENGER.superTokenInitDeploySalts(addr));
+        ghost_totalSupplyAcrossChains.set(MESSENGER.superTokenInitDeploySalts(addr), currentValue + amount);
+    }
+
+    // TODO: will need rework after
+    //   - non-atomic bridge
+    //   - `convert`
+    /// @custom:property-id 24
+    /// @custom:property sum of supertoken total supply across all chains is always equal to convert(legacy, super)-
+    /// convert(super, legacy)
+    /// @notice deliberately not a view method so medusa runs it but not the view methods defined by Test
+    function property_totalSupplyAcrossChainsEqualsMints() external {
+        // iterate over unique deploy salts aka supertokens that are supposed to be compatible with each other
+        for (uint256 deploySaltIndex = 0; deploySaltIndex < ghost_totalSupplyAcrossChains.length(); deploySaltIndex++) {
+            uint256 totalSupply = 0;
+            (bytes32 currentSalt, uint256 trackedSupply) = ghost_totalSupplyAcrossChains.at(deploySaltIndex);
+            // and then over all the (mocked) chain ids where that supertoken could be deployed
+            for (uint256 validChainId = 0; validChainId < MAX_CHAINS; validChainId++) {
+                address supertoken = MESSENGER.superTokenAddresses(validChainId, currentSalt);
+                if (supertoken != address(0)) {
+                    totalSupply += OptimismSuperchainERC20(supertoken).totalSupply();
+                }
+            }
+            assert(trackedSupply == totalSupply);
+        }
+    }
+
+    function fuzz_MockNewRemoteToken() external {
+        _deployRemoteToken();
+    }
+
+    /// @notice deploy a remote token, that supertokens will be a representation of. They are  never called, so there
+    /// is no need to actually deploy a contract for them
+    function _deployRemoteToken() internal {
+        // make sure they don't conflict with predeploys/preinstalls/precompiles/other tokens
+        remoteTokens.push(address(uint160(1000 + remoteTokens.length)));
+    }
+
+    /// @custom:property-id 14
+    /// @custom:property supertoken total supply starts at zero
```

Comment:
Ah got it. That's why you didn't use property natspec on `fuzz_DeployNewSupertoken`.
Isn't better to just declare always the property natspec on the test rather than on the function?

Reply:
I wanted to define it closest to where it's asserted so there's less of a chance it becomes outdated. Although it's true that we usually want to know which properties are asserted on by a particular _test_